temporian/implementation/numpy/operators/assign.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation for the Assign operator."""
+from typing import Dict, List
+
+import numpy as np
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+
+
+def _get_equal_sampling_indexes(
+    sampling_1: "NumpySampling", sampling_2: "NumpySampling", index: tuple
+) -> np.ndarray:
+    """Get indexes of timestamps that are equal to sampling_1 & sampling_2
+      in a specific sampling index.
+
+    Args:
+        sampling_1 (NumpySampling): sampling of assignee event.
+        sampling_2 (NumpySampling): sampling of assigned event.
+        index (tuple): index to check.
+
+    Returns:
+        np.ndarray: indexes of timestamps that are equal to sampling_1 & sampling_2.
+    """
+    return np.where(sampling_1.data[index] == sampling_2.data[index])[0]
+
+
+class NumpyAssignOperator:
+    def __call__(
+        self, assignee_event: NumpyEvent, assigned_event: NumpyEvent
+    ) -> Dict[str, NumpyEvent]:
+        """Assign features to an event.
+
+        Assignee and assigned must have same index names. Features cannot have more
+        than one row for a single index + timestamp occurence. Output event will
```

Comment:
Are we actually checking that there are no duplicated sampling values in the assigned event?

Reply:
not yet, waiting for our conversation with @getchebarne to talk about data duplication in assignee and assigned